### PR TITLE
feat: Add secure MessagBus capability

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-modbus-go/blob/master/.github/Contributing.md

## What is the current behavior?
no secure MessageBus capability

## What is the new behavior?
Secure MessageBus capability added

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by lenny-intel** [comment]
Dependent on https://github.com/edgexfoundry/device-modbus-go/pull/235 to fix build issue .

**Comment by lenny-intel** [comment]
On hold waiting for https://github.com/edgexfoundry/device-sdk-go/pull/922

**Comment by weichou1229** [comment]
> Please submit a separate PR for the Go 1.16 change

@lenny-intel I've upgraded to the v1.16 at the PR https://github.com/edgexfoundry/device-modbus-go/pull/235.

**Comment by lenny-intel** [comment]
> I've upgraded to the v1.16 at the PR #235.

@weichou1229 Thx!

File: cmd/res/configuration.toml [hunk 1]
```diff
@@ -48,9 +57,17 @@ PublishTopicPrefix = 'edgex/events' # /<device-profile-name>/<device-name>/<sour
   Retained = "false"
   AutoReconnect = "true"
   ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified  [MessageQueue.Optional]
+  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+  # Client Identifiers
+  ClientId = "device-random"
```
Comment (hahattan) [inline cmd/res/configuration.toml:63]: `device-modbus`

**Review by hahattan — CHANGES_REQUESTED** [review]
Also missing `go mod tidy` in make build recipe

**Comment by cloudxxx8** [comment]
@lenny-intel we need your update for this PR.  thanks.

File: cmd/res/configuration.toml (hunk 1 above)
Comment (lenny-intel) [reply cmd/res/configuration.toml:63]: Humm, good catch!

Comment (lenny-intel) [reply cmd/res/configuration.toml:63]: ahh, it was duplicated. Removed that extra setting.

File: Makefile [hunk 2]
```diff
@@ -16,9 +16,9 @@ GIT_SHA=$(shell git rev-parse HEAD)
 GOFLAGS=-ldflags "-X github.com/edgexfoundry/device-modbus-go.Version=$(VERSION)"
 
 build: $(MICROSERVICES)
-	$(GOCGO) build ./...
```
Comment (cloudxxx8) [inline Makefile:19]: why do we need to remove this line?  If we need to, should we also remove from 
https://github.com/edgexfoundry/device-virtual-go/pull/204/files

Comment (lenny-intel) [reply Makefile:19]: @cloudxxx8,  It causes duplicate build. Yes, I thought I had removed those when I added the `go mod tidy`.

Comment (cloudxxx8) [reply Makefile:19]: it is still there
https://github.com/edgexfoundry/device-virtual-go/blob/702ee4ef1a6f7523fa240909d56a13b89c8d46ac/Makefile#L18

**Pull request merged**